packages/amplication-build-manager/src/build-job-handler/build-job-handler.service.ts

```diff
@@ -33,23 +51,29 @@ export class BuildJobsHandlerService {
     } = dsgResourceData;
 
     const jobs: ResourceTuple[] = [];
-    if (generateServer) {
-      const serverDSGResourceData: DSGResourceData = cloneDeep(dsgResourceData);
-      serverDSGResourceData.resourceInfo.settings.adminUISettings.generateAdminUI =
-        false;
-      const jobBuildId: JobBuildId<BuildId> = `${buildId}-${EnumDomainName.Server}`;
-      await this.setJobStatus(jobBuildId, EnumJobStatus.InProgress);
-      jobs.push([jobBuildId, serverDSGResourceData]);
-    }
-
-    if (generateAdminUI) {
-      const adminUiDSGResourceData: DSGResourceData =
-        cloneDeep(dsgResourceData);
-      adminUiDSGResourceData.resourceInfo.settings.serverSettings.generateServer =
-        false;
-      const jobBuildId: JobBuildId<BuildId> = `${buildId}-${EnumDomainName.AdminUI}`;
-      await this.setJobStatus(jobBuildId, EnumJobStatus.InProgress);
-      jobs.push([jobBuildId, adminUiDSGResourceData]);
+    if (shouldSplitBuild) {
```

Comment:
this if/else is redundant. we should treat the build as an array of builds. so it can push one job or split jobs. which means instead of if/else it create a buildId array and run it accordingly

Reply:
I also think the tests should be changed accordingly

Reply:
That is indeed the logic you are mentioning as we need to add either one job per domain (server or admin-ui) or a unique one.
Am I missing something @barshimi ?

Reply:
instead of having if/else soup that can be grow in future because of parallel builds. I suggest that you'll create an array of buildIds that will trigger the job itself. so you'll be able to create one job with/without admin or `n` jobs

Reply:
Resolved offline. Bottomline, when add a new split job in the logic, we will review and update this.